### ISSUE TITLE
Define new filter.

### ISSUE THREAD

**Issue by shadrackkibet** [africanmathsinitiative/R-Instat#3897]
In this this dialog do we need this type of selector with **Data options** ?  It is causing a BUG when you click on Data options.

![image](https://user-images.githubusercontent.com/26301973/30169778-07874b12-93f6-11e7-8411-d8109d685caa.png)

**Comment by dannyparsons** [comment]
I think the Data Options should be enabled because in future we will have options for which columns to display in the selector so this will be useful.

Without any details of the bug I can't tell what the cause is or how to fix it.

**Comment by shadrackkibet** [comment]
I think this is happening because when you click on Data options in **Define New filter selector** we are opening the same dialog which is already opened and it is just below **Define New filter dialog**.

**Comment by rdstern** [comment]
Ok, this looks a good one to find.  I tried the Filter dialogue in  2 ways and (correctly) the data options are not enabled.  Goodie.

But then I defined a new filter and that gives me the sub-dialogue and there the data option is enabled.  So I clicked on it and am now an unhappy bunny!  I assume it needs to be disabled here too.

There is one interesting feature of the filter sub-dialogue that I wonder about. I define a new filter with a particular data frame diamonds.  It started with efc open and I changed to diamonds to define a filter for the diamonds.  I then clickde to define a new filter and the sub-dialogue gave me the efc dataframe again.  I can change it diamonds, but does this make any sense?

If I am doing a filter for diamonds then shouldn't the columns only be for the diamonds on the sub-dialogue?
